control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -516,8 +526,8 @@ func recordsForService(awsEndpointService *hyperv1.AWSEndpointService, hcp *hype
 
 	// If the kas is exposed through a route, the router needs to have DNS entries for both
 	// the kas and the apps domain
-	if m := servicePublishingStrategyByType(hcp, hyperv1.APIServer); m != nil && m.Type == hyperv1.Route {
-		return []string{"api", "*.apps"}
+	if m := util.ServicePublishingStrategyByTypeForHCP(hcp, hyperv1.APIServer); m != nil && m.Type == hyperv1.Route {
```

Comment:
To clarify, Is this changed unrelated to OVN but sneaked into the PR? Also is this obsiliting comment "and the apps domain"?

Reply:
Using the util function is not technically related to this PR, I just was at it so I changed that to use the common one rather than re-defining it.

The apps domain is still there, I just moved `routerDomain` to a const to be able to use it in the `RouterZoneName` func